Shopify/cli
# [Bug] [WIP] Fix importing graphql files

### WHY are these changes introduced?
The CLI is not adding the GraphQL plugin to esbuild.  In this PR I fix some code to ensure that the Graphql plugin gets added.

### WHAT is this pull request doing?
This bug was happening because:

1. require will not be available when `"type": "module"` is in the package.json
2. We wrap `require.resolve('graphql')` in a  try catch`.  This was supposed to catch when `graphql` and `graphql-tag` dependencies are not available, but because `require` is not available `isGraphqlPackageAvailable()` always returns false.
3. Therefore we never add the graphql plugin from `@luckycatfactory/esbuild-graphql-loader`

There are 2 commits:

1. First we introduce a graphql file in an extension fixture.  This is our reproduceable test case.
2. Then we add some code to ensure that require is available in the module format.  [This stackoverflow issue](https://stackoverflow.com/questions/31931614/require-is-not-defined-node-js) has more context.

### How to test your changes?

Checkout the first commit and run `yarn dev`, `yarn build` or `yarn deploy`.  The command should fail with output like this:

<img width="750" alt="Screenshot 2022-11-25 at 4 55 27 PM" src="https://user-images.githubusercontent.com/690791/204058545-329ecfdb-3465-45a0-b402-20379387c7fd.png">

Now checkout the second commit and run `yarn dev`, `yarn build` or `yarn deploy`. The command should succeed.

### Post-release steps

Should this be added to the changelog? I'm not sure.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
